sound/soc/sof/ipc4-topology.c

```diff
@@ -1547,9 +1568,37 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
-	ret = sof_ipc4_init_output_audio_fmt(sdev, &copier_data->base_config, available_fmt, ret);
+	/* set the reference params for output format selection */
+	switch (swidget->id) {
+	case snd_soc_dapm_aif_in:
+	case snd_soc_dapm_dai_out:
+	case snd_soc_dapm_buffer:
+	{
+		struct sof_ipc4_audio_format *in_fmt;
+
+		in_fmt = &available_fmt->input_pin_fmts[ret].audio_fmt;
```

Comment:
This is a bit had to follow. Why are we passing "ret" as array index here? If it has semantics (selected input format), I'd rather use more explicit variable name than generic "ret" here.